src/main/kotlin/org/rust/ide/idea/RsModuleBuilder.kt

```diff
@@ -9,29 +9,34 @@ import com.intellij.execution.ExecutionException
 import com.intellij.ide.util.projectWizard.ModuleBuilder
 import com.intellij.ide.util.projectWizard.ModuleWizardStep
 import com.intellij.ide.util.projectWizard.WizardContext
-import com.intellij.openapi.Disposable
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.module.ModuleType
 import com.intellij.openapi.options.ConfigurationException
 import com.intellij.openapi.projectRoots.SdkTypeId
 import com.intellij.openapi.roots.ModifiableRootModel
-import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.roots.ui.configuration.ModulesProvider
 import org.rust.cargo.project.settings.ui.RustProjectSettingsPanel
 import org.rust.cargo.toolchain.RustToolchain
+import org.rust.ide.idea.RsBuilderType.*
 
 /**
  * Builder which is used when a new project or module is created and not imported from source.
  */
-class RsModuleBuilder : ModuleBuilder() {
+class RsModuleBuilder(val builderType: RsBuilderType) : ModuleBuilder() {
 
     override fun getModuleType(): ModuleType<*>? = RsModuleType.INSTANCE
-
     override fun isSuitableSdkType(sdkType: SdkTypeId?): Boolean = true
+    override fun getName() = builderType.builderName
+    override fun getPresentableName() = builderType.builderName
+    override fun getDescription(): String = builderType.description
+    override fun getGroupName(): String = RsTemplatesFactory.RUST_GROUP_NAME
+    override fun isTemplateBased(): Boolean = true
+    // make builder created by RsModuleType not available
+    // to prevent additional template in new project wizard
+    override fun isAvailable(): Boolean = builderType != ROOT
 
-    override fun getCustomOptionsStep(context: WizardContext, parentDisposable: Disposable): ModuleWizardStep =
-        CargoConfigurationWizardStep(context).apply {
-            Disposer.register(parentDisposable, Disposable { this.disposeUIResources() })
-        }
+    override fun createWizardSteps(wizardContext: WizardContext, modulesProvider: ModulesProvider): Array<ModuleWizardStep> =
+        arrayOf(CargoConfigurationWizardStep(wizardContext))
```

Comment:
I'm not sure how to register `Disposable` here because IDEA doesn't pass `parentDisposable` into `createWizardSteps` method.